impl/src/main/java/org/jboss/weld/util/Beans.java

```diff
@@ -223,16 +224,22 @@ public static boolean containsAllInterceptionBindings(Set<Annotation> expectedBi
 
     public static boolean isBeanEnabled(Bean<?> bean, ModuleEnablement enabled) {
         if (bean.isAlternative()) {
+            boolean isEnabled = false;
             if (enabled.isEnabledAlternativeClass(bean.getBeanClass())) {
-                return true;
+                isEnabled = true;
             } else {
                 for (Class<? extends Annotation> stereotype : bean.getStereotypes()) {
                     if (enabled.isEnabledAlternativeStereotype(stereotype)) {
-                        return true;
+                        isEnabled = true;
+                        break;
                     }
                 }
-                return false;
             }
+            // For synthetic enabled alternatives, the ModuleEnablement may not yet be aware of them
+            if (!isEnabled && bean instanceof WeldBean && ((WeldBean<?>) bean).getPriority() != null) {
```

Comment:
Hm, how about custom beans implementing `jakarta.enterprise.inject.spi.Prioritized`? I'm just curious ;-)

Reply:
Good point, it is probably affected as well. I'll check it

Reply:
@mkouba how about now? :)